cedar-lean/Cedar/Thm/Partial/Evaluation/GetAttr.lean

```diff
@@ -0,0 +1,126 @@
+/-
+ Copyright Cedar Contributors
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-/
+
+import Cedar.Partial.Evaluator
+import Cedar.Spec.Evaluator
+import Cedar.Thm.Data.Control
+import Cedar.Thm.Data.LT
+import Cedar.Thm.Data.Map
+import Cedar.Thm.Data.Set
+import Cedar.Thm.Partial.Evaluation.Basic
+
+namespace Cedar.Thm.Partial.Evaluation.GetAttr
+
+open Cedar.Data
+open Cedar.Spec (Attr EntityUID Error Result)
+
+/--
+  if `entities.attrs uid` is `ok` with some attrs, those attrs are a
+  well-formed `Map`
+-/
+theorem partialEntities_attrs_wf {entities : Partial.Entities} {uid : EntityUID} {attrs: Map String Partial.Value} :
+  entities.AllWellFormed →
+  entities.attrs uid = .ok attrs →
+  attrs.WellFormed
+:= by
+  unfold Partial.Entities.attrs Partial.Entities.AllWellFormed Partial.EntityData.WellFormed
+  intro wf h₁
```

Comment:
Minor style comment---definitely no need to change this anytime soon:  hypotheses names start with `h` in all the Lean code I've looked at (e.g., Std).  It's unusual to see one that doesn't, except for `ih`.  I tend to write `hwf` for the well-formedness ones.